controllers/metal3.io/hostfirmwarecomponents_test.go

```diff
@@ -0,0 +1,331 @@
+			LastVersionFlashed: "1.1.0",
+		},
+		{
+			Component:      "bios",
+			InitialVersion: "1.0.1",
+			CurrentVersion: "1.0.1",
+		},
+	}
+	return components
+}
+
+// Create the baremetalhost reconciler and use that to create bmh in same namespace.
+func createBaremetalHostHFC() *metal3api.BareMetalHost {
+	bmh := &metal3api.BareMetalHost{}
+	bmh.ObjectMeta = metav1.ObjectMeta{Name: "hostName", Namespace: "hostNamespace"}
+	c := fakeclient.NewFakeClient(bmh)
+
+	reconciler := &BareMetalHostReconciler{
+		Client:             c,
+		ProvisionerFactory: nil,
+		Log:                ctrl.Log.WithName("bmh_reconciler").WithName("BareMetalHost"),
+	}
+	reconciler.Create(context.TODO(), bmh)
+
+	return bmh
+}
+
+// Create and HFC with input spec components.
+func getHFC(spec metal3api.HostFirmwareComponentsSpec) *metal3api.HostFirmwareComponents {
+	hfc := &metal3api.HostFirmwareComponents{}
+
+	hfc.Status = metal3api.HostFirmwareComponentsStatus{
+		Components: []metal3api.FirmwareComponentStatus{
+			{
+				Component:      "bmc",
+				InitialVersion: "1.0.0",
+				CurrentVersion: "1.0.0",
+			},
+			{
+				Component:      "bios",
+				InitialVersion: "1.0.1",
+				CurrentVersion: "1.0.1",
+			},
+		},
+	}
+
+	hfc.TypeMeta = metav1.TypeMeta{
+		Kind:       "HostFirmwareComponents",
+		APIVersion: "metal3.io/v1alpha1"}
+	hfc.ObjectMeta = metav1.ObjectMeta{
+		Name:      hostName,
+		Namespace: hostNamespace}
+
+	hfc.Spec = spec
+	return hfc
+}
+
+// Test the hostfirmwarecomponents reconciler functions.
+func TestStoreHostFirmwareComponents(t *testing.T) {
+	testCases := []struct {
+		Scenario           string
+		CurrentHFCResource *metal3api.HostFirmwareComponents
+		ExpectedComponents *metal3api.HostFirmwareComponents
+	}{
+		{
+			Scenario: "updated components",
+			CurrentHFCResource: &metal3api.HostFirmwareComponents{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "HostFirmwareComponents",
+					APIVersion: "metal3.io/v1alpha1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "hostName",
+					Namespace:       "hostNamespace",
+					ResourceVersion: "1"},
+				Spec: metal3api.HostFirmwareComponentsSpec{
+					Updates: []metal3api.FirmwareUpdate{
+						{
+							Component: "bmc",
+							URL:       "https://myurl/newbmcfirmware",
+						},
+					},
+				},
+				Status: metal3api.HostFirmwareComponentsStatus{
+					Updates: []metal3api.FirmwareUpdate{
+						{
+							Component: "bmc",
+							URL:       "https://myurls/newbmcfirmware",
+						},
+					},
+					Components: []metal3api.FirmwareComponentStatus{
+						{
+							Component:          "bmc",
+							InitialVersion:     "1.0.0",
+							CurrentVersion:     "1.1.0",
+							LastVersionFlashed: "1.1.0",
+						},
+						{
+							Component:      "bios",
+							InitialVersion: "1.0.1",
+							CurrentVersion: "1.0.1",
+						},
+					},
+				},
+			},
+			ExpectedComponents: &metal3api.HostFirmwareComponents{
+				Spec: metal3api.HostFirmwareComponentsSpec{
```

Comment:
nit: I don't think you use this Spec anywhere

Reply:
Now I'm using, added to the test =)